test/test.lua

```diff
@@ -34,6 +34,16 @@ local function isEqual(a, b, tolerance, ...)
    end
 end
 
+local function checkMultiDevice(x, fn, ...)
```

Comment:
What's the purpose of this? Why are we running each function again on a different device?